mono/metadata/object.c

```diff
@@ -3955,36 +3955,46 @@ mono_runtime_get_main_args (void)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 	MonoError error;
-	MonoArray *result = mono_runtime_get_main_args_checked (&error);
+	MonoArrayHandle arg_array = mono_runtime_get_main_args_handle (&error);
```

Comment:
This leaks a handle.  This function should do `HANDLE_FUNCTION_ENTER`/`HANDLE_FUNCTION_RETURN_OBJ (result)`.

The `HANDLE_FUNCTION_RETURN_OBJ` macro takes a handle and then returns the raw pointer of that handle.